### PR TITLE
Separate maintenance content into a maintenance doc.

### PR DESCRIPTION
**Pull request by rdblue**

### CODE REVIEW

File: site/mkdocs.yml [hunk 1]
```diff
@@ -49,8 +49,9 @@ nav:
   - User docs:
     - Getting Started: getting-started.md
     - Spark: spark.md
-    - Spark Structured Streaming: spark-structured-streaming.md
+    - Spark Streaming: spark-structured-streaming.md
```
Comment (HeartSaVioR) [inline site/mkdocs.yml:52]: I think it's important for Spark users to "specify" whether it is DStream or Structured Streaming. They often mixed up both which brings confusion.

File: site/mkdocs.yml [hunk 2]
```diff
@@ -49,8 +49,9 @@ nav:
...
     - Presto: presto.md
+    - Maintenance: maintenance.md
```
Comment (HeartSaVioR) [inline site/mkdocs.yml:54]: Looks like the page is missing in this PR - probably miss to add a new file?

File: site/mkdocs.yml (hunk 1 above)
Comment (rdblue) [reply site/mkdocs.yml:52]: I think it's okay to use the short form in menu navigation. This doesn't change the actual page title, right?

File: site/mkdocs.yml (hunk 2 above)
Comment (rdblue) [reply site/mkdocs.yml:54]: Fixed.

File: site/mkdocs.yml (hunk 1 above)
Comment (rdblue) [reply site/mkdocs.yml:52]: Confirmed locally that it doesn't change the page title. Are you okay using a shorter name in the nav?

Comment (HeartSaVioR) [reply site/mkdocs.yml:52]: Ah OK that's short form on the menu. Looks great.

File: site/docs/maintenance.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+Manifests in the metadata tree are automatically compacted in the order they are added, which makes queries faster when the write pattern aligns with read filters. For example, writing hourly-partitioned data as it arrives is aligned with time range query filters.
+
+When a table's write pattern doesn't align with the query pattern, metadata and be rewritten to re-group data files into manifests using `rewriteManifests` or the `rewriteManifests` action (for parallel rewrites using Spark).
+
+This example rewrites small manifests and groups data files by the first partition field.
+
+```java
+Table table = ...
+table.rewriteManifests()
+    .rewriteIf(file -> file.length() < 4 * 1024 * 1024) // 10 MB
```
Comment (HeartSaVioR) [inline site/docs/maintenance.md:126]: nit: the code represents 4 MB, whereas the comment represents 10 MB

**Review by HeartSaVioR — APPROVED** [review]
Looks great, just a nit which I can change after merging.

**Pull request merged**